nebula/api.go

```diff
@@ -104,18 +104,27 @@ func (c *APIClient) TryUpdate(siteName string, hostID string, privateKey string,
 	// Check for update
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
-	updateAvailable, err := c.c.CheckForUpdate(ctx, creds)
+	msg, err := c.c.LongPollWait(ctx, creds, []string{"DoUpdate"})
 	switch {
 	case errors.As(err, &dnapi.ErrInvalidCredentials):
 		return nil, InvalidCredentialsError{}
 	case err != nil:
 		return nil, fmt.Errorf("CheckForUpdate error: %s", err)
 	}
-
-	if !updateAvailable {
+	var msgType struct{ Command string }
+	err = json.Unmarshal(msg.Action, &msgType)
```

Comment:
Dang we return a raw JSON from dnapi for this? That's kinda ugly.

Reply:
strong types across the language boundary is kinda poorly supported, and without the alignment patch, would crash sometimes :cry: